client_it_test.go

```diff
@@ -36,14 +36,55 @@ import (
 	"github.com/hazelcast/hazelcast-go-client/hzerrors"
 	"github.com/hazelcast/hazelcast-go-client/internal"
 	"github.com/hazelcast/hazelcast-go-client/internal/it"
-	"github.com/hazelcast/hazelcast-go-client/internal/proxy"
+	"github.com/hazelcast/hazelcast-go-client/internal/it/skip"
 	"github.com/hazelcast/hazelcast-go-client/logger"
 	"github.com/hazelcast/hazelcast-go-client/types"
 )
 
-var idGen = proxy.ReferenceIDGenerator{}
+func TestClient(t *testing.T) {
+	testCases := []struct {
```

Comment:
We can add a new field such as `isParallel` as a bool, then change the test cases loop like;
```
for _, tc := range testCases {
	t.Run(tc.name, func(t *testing.T) {
		if tc.isParrallel {
			t.Parallel()
			tc.f(t)
		} else {
			tc.f(t)
		}
	})
}
```
Then, we can remove t.Parallel() call from each of the task. By doing that, we can isolate a single test by setting its parallel bool field.

Reply:
Yes, I use that in the map tests, where some of the tests should not be parallelized: https://github.com/hazelcast/hazelcast-go-client/blob/18f98951fbc00c0492ba3f92dae6b39bb88ec8e9/map_it_test.go#L49